src/uproot/models/RNTuple.py

```diff
@@ -218,26 +220,62 @@ def page_list_envelopes(self):
 
         return self.column_innerlist_dict
 
-    def col_form(self, field_id):
-        for cr in self.header.column_records:
+    def base_col_form(self, col_id):
+        cr = self.header.column_records[col_id]
+        form_key = f"column-{col_id}"
+        if cr.type == 3:  # switch (UnionForm)
+            return form_key
+        elif cr.type > 2:  # data column
+            return ak._v2.forms.NumpyForm(
+                _rntuple_col_types[cr.type], form_key=form_key
+            )
+        else:  # offset index column
+            return form_key
+
+    def col_form(self, field_id, type_name=None):
+        # FIXME remove this ugly logic
+        rel_crs = []
+        rel_crs_idxs = []
+        for i, cr in enumerate(self.header.column_records):
             if cr.field_id == field_id:
-                form_key = f"field-{cr.field_id}"
-                if cr.type > 2:  # data column
-                    return ak._v2.forms.NumpyForm(
-                        _rntuple_col_types[cr.type], form_key=form_key
-                    )
-                else:  # offset index column
-                    return form_key
+                rel_crs.append(cr)
+                rel_crs_idxs.append(i)
+            if cr.field_id > field_id:
+                break
+        if len(rel_crs) == 1:  # normal case
+            form_key = f"column-{rel_crs_idxs[0]}"
+            cr = rel_crs[0]
+            if cr.type == 3:  # switch (UnionForm)
+                return form_key
+            elif cr.type > 2:  # data column
+                return ak._v2.forms.NumpyForm(
+                    _rntuple_col_types[cr.type], form_key=form_key
+                )
+            else:  # offset index column
+                return form_key
+        elif type_name == "std::string":  # string field splits->2 in col records
+            assert len(rel_crs_idxs) == 2
+            cr_char = rel_crs[-1]
+            assert cr_char.type == 5  # char
+            inner = self.base_col_form(rel_crs_idxs[-1])
+            form_key = f"column-{rel_crs_idxs[0]}"
+            return ak._v2.forms.ListOffsetForm(
+                "u32", inner, form_key=form_key, parameters={"__array__": "string"}
+            )
```

Comment:
Perhaps you didn't get an error, but the ListOffsetForm has to have `parameters={"__array__": "string"}` _and also_ the NumpyForm has to have `parameters={"__array__": "char"}`. It's a non-local requirement (an invariant that applies to sets of nodes, rather than a single node), so it's not checked in the node constructor, but [ak.is_valid](https://awkward-array.readthedocs.io/en/latest/_auto/ak.is_valid.html) checks for it.